parsl/providers/aws/aws.py

```diff
@@ -648,11 +649,15 @@ def submit(self, command='sleep 1', tasks_per_node=1, job_name="parsl.aws"):
             If at capacity, None will be returned. Otherwise, the job identifier will be returned.
         """
 
-        job_name = self.generate_aws_id()
-        wrapped_cmd = self.launcher(command,
-                                    tasks_per_node,
-                                    self.nodes_per_block)
-        [instance, *rest] = self.spin_up_instance(command=wrapped_cmd, job_name=job_name)
+        try:
+            job_name = self.generate_aws_id()
+            wrapped_cmd = self.launcher(command,
+                                        tasks_per_node,
+                                        self.nodes_per_block)
+            [instance, *rest] = self.spin_up_instance(command=wrapped_cmd, job_name=job_name)
+        except Exception as e:
+            raise SubmitException(job_name,
```

Comment:
I think the better option here would be to add an `logger.exception(...DESCRIPTION..)` followed by a `raise` so that the exception is logged and left to the layer above to deal with.